ote/src/clj/ote/services/routes.clj

```diff
@@ -89,19 +91,33 @@
                 (fn [{::transit/keys [code] :as stop}]
                   (or (custom-stops code) stop))))))
 
-(defn save-route [nap-config db user route]
-  (authorization/with-transport-operator-check
-    db user (::transit/transport-operator-id route)
-    (fn []
-      (tx/with-transaction db
-        (let [r (-> route
-                    (save-custom-stops db user)
-                    (modification/with-modification-fields ::transit/id user)
-                    (update ::transit/stops #(mapv stop-location-geometry %))
-                    (update ::transit/service-calendars #(mapv service-calendar-dates->db %))
-                    (update ::transit/trips (flip mapv) transit/trip-stop-times-to-24h))]
-          (log/debug "Save route: " r)
-          (upsert! db ::transit/route r))))))
+(defn- stop-time->pginterval [stop-time]
+  (cond-> stop-time
+          (some? (::transit/departure-time stop-time)) (update ::transit/departure-time #(ote.time/->PGInterval %))
+          (some? (::transit/arrival-time stop-time)) (update ::transit/arrival-time #(ote.time/->PGInterval %))))
+
+(defn- sort-tripsv [trips]
+  (vec (sort-by
+         (juxt
+           (comp ote.time/minutes-from-midnight ote.time/pginterval->interval ::transit/departure-time first ::transit/stop-times)
+           (comp ote.time/minutes-from-midnight ote.time/pginterval->interval ::transit/arrival-time last ::transit/stop-times))
+         trips)))
+
+(defn sort-stop-times-of-tripsv [trips]
+  (mapv (fn [trip]
+          (update trip ::transit/stop-times #(vec (sort-by ::transit/stop-idx %))))
+        trips))
+
+(defn fetch-sea-trips [db route-id]
+  (-> (fetch db
+             ::transit/trip
+             (conj (specql/columns ::transit/trip)
+                   [::transit/stop-times  ; Nests linked stop-time table records under each trip record
+                    (specql/columns ::transit/stop-time)])
+             {:transit-trip/route-id route-id})
+      ; Trips must be in same order as stops because of front-end design so sorted just in case.
+      sort-stop-times-of-tripsv
```

Comment:
Check if this could be done using specql/sort-by

Reply:
Couldn't get this working for _nested_ joined table, works on first level tables.